packages/web/src/components/apps/use-package-json.tsx

```diff
@@ -1,13 +1,20 @@
-import React, { createContext, useCallback, useContext, useState } from 'react';
+import React, { createContext, useCallback, useContext, useEffect, useState } from 'react';
 import { OutputType } from '@srcbook/components/src/types';
 import { AppChannel } from '@/clients/websocket';
-import { DepsInstallLogPayloadType, DepsInstallStatusPayloadType } from '@srcbook/shared';
+import {
+  DepsInstallLogPayloadType,
+  DepsInstallStatusPayloadType,
+  DepsStatusResponsePayloadType,
+} from '@srcbook/shared';
 import { useLogs } from './use-logs';
 
 type NpmInstallStatus = 'idle' | 'installing' | 'complete' | 'failed';
 
 export interface PackageJsonContextValue {
-  npmInstall: (packages?: string[]) => void;
+  npmInstall: (packages?: string[]) => Promise<void>;
+  clearNodeModules: () => void;
+
+  nodeModulesExists: boolean | null;
   status: NpmInstallStatus;
   installing: boolean;
   failed: boolean;
```

Comment:
Something I really am not a fan of here: `nodeModulesExists` is `boolean | null` - it is null before it is set by the server initially.

What sucks about this is that any check of this value must explicitly check `=== true`, `=== false`, etc since both `null` and `false` are falsey.

I wanted to get this out to minimize merge conficts. But, I think in hindsight this should have been an enum like `'loading' | 'enabled' | 'disabled'` or something like that. Feel free to make that change, or if not, I'll probablhy make it on monday!

Reply:
One of my good mentors said that nearly all booleans should just be enums :), agree we should switch it. Not super urgent but minor tech debt in its current state